main.go

```diff
@@ -10,23 +10,47 @@ import (
 	"github.com/bitrise-io/go-utils/log"
 	"github.com/bitrise-io/go-utils/pathutil"
 	"github.com/bitrise-io/go-utils/sliceutil"
+	"github.com/bitrise-steplib/bitrise-step-android-unit-test/cache"
 	"github.com/bitrise-tools/go-android/gradle"
 	"github.com/bitrise-tools/go-steputils/stepconf"
+	shellquote "github.com/kballard/go-shellquote"
 )
 
 // Config ...
 type Config struct {
-	ProjectLocation   string `env:"project_location,required"`
+	ProjectLocation   string `env:"project_location,dir"`
 	ReportPathPattern string `env:"report_path_pattern"`
 	Variant           string `env:"variant"`
 	Module            string `env:"module"`
+	Arguments         string `env:"arguments"`
+	CacheLevel        string `env:"cache_level,opt[none,only_deps,all]"`
 }
 
 func failf(f string, args ...interface{}) {
 	log.Errorf(f, args...)
 	os.Exit(1)
 }
 
+func getArtifacts(gradleProject gradle.Project, started time.Time, pattern string) (artifacts []gradle.Artifact, err error) {
+	for _, t := range []time.Time{started, time.Time{}} {
```

Comment:
this is problematic, since the startTime may equals to the empty time.

alternative:

```
func getArtifacts1(started time.Time, pattern string) []string {
	artifacts := []string{} // find artifacts
	if len(artifacts) > 0 {
		return artifacts
	}

	artifacts = []string{} // fallback: find artifacts
	if len(artifacts) == 0 {

	}
}
```